src/main/java/me/xemor/configurationdata/PotionEffectData.java

```diff
@@ -8,37 +9,34 @@
 
 public class PotionEffectData {
 
-    private PotionEffect potionEffect = null;
-
-    public PotionEffectData(ConfigurationSection configurationSection) {
-        PotionEffectType potionType = PotionEffectType.getByName(configurationSection.getString("type", "d").toUpperCase());
-        if (potionType == null) {
-            ConfigurationData.getLogger().severe("Invalid potion effect type specified at " + configurationSection.getCurrentPath() + ".type");
-            return;
-        }
-        int potency = configurationSection.getInt("potency");
-        if (potency > 0) {
-            potency--;
-        }
-        boolean ambient = configurationSection.getBoolean("ambient", true);
-        boolean hasParticles = configurationSection.getBoolean("hasParticles", true);
-        double duration = configurationSection.getDouble("duration", 5);
-        createPotion(potionType, duration, potency, ambient, hasParticles);
+    @JsonProperty
+    private PotionEffectType type;
+    @JsonProperty
+    private double duration = 5;
+    @JsonProperty
+    private int potency = 0;
+    @JsonProperty
+    private boolean ambient = true;
+    @JsonProperty
+    private boolean hasParticles = true;
+
+    public PotionEffectData() {
+        if (potency > 0) potency--;
     }
 
-    public Optional<PotionEffect> getPotionEffect() {
-        return Optional.ofNullable(potionEffect);
+    public PotionEffect getPotionEffect() {
+        return createPotion();
     }
 
-    protected void createPotion(PotionEffectType type, double duration, int potency, boolean ambient, boolean hasParticles) {
+    private PotionEffect createPotion() {
         if (type.isInstant()) {
```

Comment:
My personal code preference here would be something like this, just thought I'd suggest, feel free to ignore this if it's not of interest,
```suggestion
        if (type.isInstant()) {
            duration = 1;
        } else if (duration != 0) {
            duration = (int) Math.round(duration * 20);
        } else {
            duration = Integer.MAX_VALUE;
        }
        
        return new PotionEffect(type, duration, potency, ambient, hasParticles);
```
Or you could even smush it into a messy one liner
```suggestion
        duration = type.isInstant() ? 1 : duration != 0 ? (int) Math.round(duration * 20) : Integer.MAX_VALUE;
        return new PotionEffect(type, duration, potency, ambient, hasParticles);
```

Reply:
yep ok